internal/ast/ast.go

```diff
@@ -10466,3 +10492,361 @@ type PragmaSpecification struct {
 func (spec *PragmaSpecification) IsTripleSlash() bool {
 	return (spec.Kind & PragmaKindTripleSlashXML) > 0
 }
+
+// JS syntactic diagnostics functions
+
+func (sourceFile *SourceFile) getJSSyntacticDiagnosticsForFile() []*Diagnostic {
+	var diagnostics []*Diagnostic
+
+	// Walk the entire AST to find TypeScript-only constructs
+	sourceFile.walkNodeForJSDiagnostics(sourceFile.AsNode(), sourceFile.AsNode(), &diagnostics)
+
+	return diagnostics
+}
+
+func (sourceFile *SourceFile) walkNodeForJSDiagnostics(node *Node, parent *Node, diags *[]*Diagnostic) {
```

Comment:
Move this code to a new file in the `ast` package, and make these regular functions, not methods.